docs/diagrams/DeleteClassSequence.puml

```diff
@@ -0,0 +1,76 @@
+@startuml
+!include style.puml
+skinparam ArrowFontStyle plain
+!define input "/delete_class module/CS2103T tutorial/T09"
+
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":AddressBookParser" as AddressBookParser LOGIC_COLOR
+participant ":DeleteClassCommandParser" as DeleteClassCommandParser LOGIC_COLOR
+participant "d:DeleteClassCommand" as DeleteClassCommand LOGIC_COLOR
+participant "r:CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "m:Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute(input)
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand(input)
+activate AddressBookParser
+
+create DeleteClassCommandParser
+AddressBookParser -> DeleteClassCommandParser
+activate DeleteClassCommandParser
+
+DeleteClassCommandParser --> AddressBookParser
+deactivate DeleteClassCommandParser
+
+AddressBookParser -> DeleteClassCommandParser : parse("module/CS2103T tutorial/T09")
+activate DeleteClassCommandParser
+
+create DeleteClassCommand
+DeleteClassCommandParser -> DeleteClassCommand
+activate DeleteClassCommand
+
+DeleteClassCommand --> DeleteClassCommandParser :
+deactivate DeleteClassCommand
+
+DeleteClassCommandParser --> AddressBookParser : d(p)
+deactivate DeleteClassCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+DeleteClassCommandParser -[hidden]-> AddressBookParser
+destroy DeleteClassCommandParser
+
+AddressBookParser --> LogicManager : d(p)
+deactivate AddressBookParser
+
+LogicManager -> DeleteClassCommand : execute(m)
+activate DeleteClassCommand
+
+DeleteClassCommand -> Model : findModuleFromList("CS2103T")
+activate Model
+
+Model --> DeleteClassCommand : existingModule
+DeleteClassCommand -> Model : existingModule.deleteTutorialClass("T09"")
+deactivate Model
+
+DeleteClassCommand --> DeleteClassCommand : deleteClass(existingModule, "T09")
```

Comment:
I think you might need to activate a new activation bar for deleteclasscommand when self-calling methods since the activation bar seems to be cutoff at the wrong location.

Reply:
ok have edited to deactive deleteclasscommand later to extend the activation bar